src/common-verifier/src/main/java/ee/ria/xroad/common/ocsp/OcspVerifier.java

```diff
@@ -184,11 +192,12 @@ private synchronized SingleResp verifyResponseValidityCached(OCSPResp response,
                                                                  X509Certificate issuer)
             throws Exception {
         String key = SINGLE_RESP + response.hashCode() + subject.hashCode() + issuer.hashCode();
-        if (!CACHE.isValid(key)) {
-            CACHE.setValue(key, verifyResponseValidity(response, subject, issuer));
+        SingleResp responseValidity = RESPONSE_VALIDITY_CACHE.getIfPresent(key);
+        if (responseValidity == null) {
+            responseValidity = verifyResponseValidity(response, subject, issuer);
+            RESPONSE_VALIDITY_CACHE.put(key, responseValidity);
```

Comment:
Suggestion: use the cache.get(key, loader). It also makes it possible to remove the `synchronized` modifier. something like
```
try {
   final String key = ...
   return RESPONSE_VALIDITY_CACHE.get(key, () -> verifyResponseValidity(response, subject, issuer));
 } catch (ExecutionException | UncheckedExecutionException e) {
   throw (Exception)e.getCause();
 }
```